tests/Unit/Core/Form/IdentifiableObject/DataProvider/CombinationFormDataProviderTest.php

```diff
@@ -0,0 +1,251 @@
+    }
+
+    /**
+     * @dataProvider getExpectedData
+     *
+     * @param array $combinationData
+     * @param array $expectedData
+     */
+    public function testGetData(array $combinationData, array $expectedData)
+    {
+        $queryBusMock = $this->createQueryBusMock($combinationData);
+        $provider = new CombinationFormDataProvider($queryBusMock);
+
+        $formData = $provider->getData(static::COMBINATION_ID);
+        // assertSame is very important here We can't assume null and 0 are the same thing
+        $this->assertSame($expectedData, $formData);
+    }
+
+    public function getExpectedData(): Generator
+    {
+        $datasetsByType = [
+            $this->getDatasetsForStock(),
+        ];
+
+        foreach ($datasetsByType as $datasetByType) {
+            foreach ($datasetByType as $dataset) {
+                yield $dataset;
+            }
+        }
+    }
+
+    /**
+     * @return array
+     */
+    private function getDatasetsForStock(): array
+    {
+        $datasets = [];
+
+        $expectedOutputData = $this->getDefaultOutputData();
+        $combinationData = [];
+
+        $datasets[] = [
+            $combinationData,
+            $expectedOutputData,
+        ];
+
+        $expectedOutputData = $this->getDefaultOutputData();
+        $combinationData = [
+            'quantity' => 42,
+            'minimal_quantity' => 7,
+            'low_stock_threshold' => 5,
+            'low_stock_alert' => true,
+            'location' => 'top shelf',
+            'available_date' => new DateTime('1969/07/20'),
+        ];
+        $expectedOutputData['stock']['quantity'] = 42;
+        $expectedOutputData['stock']['minimal_quantity'] = 7;
+        $expectedOutputData['stock']['low_stock_threshold'] = 5;
+        $expectedOutputData['stock']['low_stock_alert'] = true;
+        $expectedOutputData['stock']['stock_location'] = 'top shelf';
+        $expectedOutputData['stock']['available_date'] = '1969-07-20';
+
+        $datasets[] = [
+            $combinationData,
+            $expectedOutputData,
+        ];
+
+        return $datasets;
+    }
+
+    /**
+     * @param array $combinationData
+     *
+     * @return MockObject|CommandBusInterface
+     */
+    private function createQueryBusMock(array $combinationData)
+    {
+        $queryBusMock = $this->createMock(CommandBusInterface::class);
+
+        $queryBusMock
+            ->method('handle')
+            ->with($this->logicalOr(
+                $this->isInstanceOf(GetCombinationForEditing::class)
+            ))
+            ->willReturnCallback(function ($query) use ($combinationData) {
+                return $this->createResultBasedOnQuery($query, $combinationData);
+            })
+        ;
+
+        return $queryBusMock;
+    }
+
+    /**
+     * @param $query
+     * @param array $combinationData
+     *
+     * @return CombinationForEditing
+     */
+    private function createResultBasedOnQuery($query, array $combinationData)
```

Comment:
```suggestion
    private function createResultBasedOnQuery($query, array $combinationData): CombinationForEditing
```

Reply:
I can't because this method is a proxy that returns multiple types based on the input query So for now it only handles one case indeed because it's the first PR
But in the next one we will have to mock another CQRS Query which will have another type of return